### ISSUE TITLE
Startup command/script for java-buildpack

### ISSUE THREAD

**Issue by ManuelUlrich** [cloudfoundry/java-buildpack#736]
Currently it is possible to use a .procfile in the binary buildpack which helps if there is a need to do some stuff before application start like setting environment variables that are only visible within the application process. It would be very beneficial to have such a feature also for the java-buildpack. Sadly .profile does not work in the java-buildpack and also the command option in the manifest is useless as it overrides the buildpack JAVA-OPS and prevents the application from starting.
 
**Scenario:**
We have a main application that expects some secrets as flat environment variables. The application is build in a generic way and should be platform independent as it also runs on Kubernetes or VM webservers besides our internal PCF. For PCF we carved out the secrets from the manifest and put them into credhub. Problem is now that they are accessible inside the JSON BLOB and the main app expects flat environment variables. Idea is now to source a script that is provided inside the jar which exports the flat environment variables from credhub via jq. This works wonderful for the binary-buildpacks but for java-buildpacks we do not have any change to call the script. Again, we would like to not change the main app as this should be independent. Our target is to fully hide the environment variables for any user that could use the AppManager or ssh on the app. Exporting the flat variables via a startup script works great as it is only visible for the running application ND and works for deployments, restarts and restages.
 
Also creating our own buildpack is not an ideal solution for us, as we are a big enterprise and it is hard to maintain many buildpacks. Also we would have to maintain another buildpack just for this use case.
 
I could think of 2 extensions to the java-buildpack that could make such a use case possible:
1. Possibility to execute a .profile inside the root directory of the jar file (referring to this article: https://github.com/cloudfoundry/docs-dev-guide/issues/133 it is not what Pivotal likes much but in my use case I cannot have the workaround of using set-env as they are visible and not suited for secrets)
2. Manifest "command" should have a feature to append commands and not replace the buildpack command. This would be a feature that I would really love as the "command" is currently useless for the java-buildpack because the cannot run without the buildpack startup commands. I also do not want to hard code the JAVA_OPTS in the manifest as I would have to specify paths and versions. Appending a source command of a script would also solve the problem.
 
What do you think?

**Comment by ManuelUlrich** [comment]
After some more testing I found out that option 1 is now working with the java-buildpack ;)

**Comment by berndgoetz** [comment]
Hi @ManuelUlrich I landed here after some poking around in the Internet - how did you achieve running the command with Java buildpack? My concrete task at hand is to parse the VCAP_SERVICES and create a new specific environment variable from one value in it provided through a bound service (a connection string variable for the Application Insights agent which requires the environment variable set even before the Spring container with all its greatness is there).
I found https://github.com/starkandwayne/cf-pancake but I'll see how quickly I can get it deployed in our infrastructure. So I'd be interested how you solved it. Thanks.
